src/Relude/List.hs

```diff
@@ -13,7 +13,34 @@ Utility functions to work with lists.
 module Relude.List
        ( module Relude.List.NonEmpty
        , module Relude.List.Reexport
+       , (!!?)
        ) where
 
+import Data.Bool (otherwise, (||))
+import Data.List (length, (!!))
+import Data.Maybe (Maybe (..))
+
+import Relude.Base ((<), (>=))
 import Relude.List.NonEmpty
 import Relude.List.Reexport
+import Relude.Numeric (Int)
+
+-- $setup
+-- >>> import Relude
+{- | Safer version of '!!', returns a Maybe.
+get element from list using index value starting from `0`.
+
+>>> [] !!? 0
+Nothing
+
+>>> ["a", "b", "c"] !!? 3
+Nothing
+
+>>> ["a", "b", "c"] !!? 2
+Just "c"
+-}
+(!!?) :: [a] -> Int -> Maybe a
```

Comment:
Let's also add fixity declaration to this operator

Reply:
I'm not really sure how to do this, I added an `infixl 9 !!?` line but what does the number represent? How do you people do this? 😂 (It's the first infix operator I define in my life 😅 )

Reply:
@kutyel The number represents the priority of operators. Or, how the `()` are placed implicitly. For example, `.` operator has priority 9 (the highest possible), and `$` has priority `0` (the lowest possible. That's why when you see code `foo $ bar . baz` you know that parenthesis are placed like this: `foo $ (bar . baz)` and not like this `(foo $ bar) . baz`.

We can check the priority for the `!!` operator and it's also 9. However, instead of `infixl` let's write just `infix` (because it's not possible to chain this operator). According to our style guide, fixity declaration should be between function documentation and the function implementation itself.